### DISCUSSION TITLE
[@types/luxon] toISODate possibly null

### DISCUSSION THREAD

**Discussion by slaesh** [DefinitelyTyped/DefinitelyTyped#65161] [Issues with a @types package] [answered]
@types/luxon v3.3.0 introduces `IfInvalid`, but it seems to be "wrong" for some of the functions.

luxon says that it will return a string for the `toISODate` function. `null` shouldnt be possible though.

![image](https://user-images.githubusercontent.com/55546823/232427638-9c03f631-1768-4eb6-a10d-357af159d7f1.png)

https://github.com/moment/luxon/issues/760

[a comment by an automated account is not shown]

**Comment by slaesh** [comment]
looking at the code, there is no return path returning `null` ..

https://github.com/moment/luxon/blob/07023283e74321c2080de9bb73070a3ffd2a7acc/src/datetime.js#L197

![image](https://user-images.githubusercontent.com/55546823/232429939-2bfa1a16-c1df-4053-8e00-46d095315bbf.png)

**Comment by slaesh** [comment]
guess this applies to mostly all function within the `datetime.d.ts`? :thinking:

**Comment by CarsonF** [comment, marked as answer]
If the DateTime was created as invalid, then yes it can return null. 
https://github.com/moment/luxon/blob/07023283e74321c2080de9bb73070a3ffd2a7acc/src/datetime.js#L1668-L1671

If you want to mark luxon to throw instead on invalid objects you can set that.
```ts
Settings.throwOnInvalid = true;

declare module 'luxon' {
  interface TSSettings {
    throwOnInvalid: true;
  }
}
```

This change is from https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64995

**Reply by slaesh** [reply]
awesome thanks, didnt saw that :( sorry!
